crypto/x509/x509_vfy.c

```diff
@@ -973,28 +995,190 @@ static int check_trust(X509_STORE_CTX *ctx, int num_untrusted)
 static int check_revocation(X509_STORE_CTX *ctx)
 {
     int i = 0, last = 0, ok = 0;
-
-    if ((ctx->param->flags & X509_V_FLAG_CRL_CHECK) == 0)
+    int crl_check_enabled =
+        (ctx->param->flags &
+        (X509_V_FLAG_CRL_CHECK | X509_V_FLAG_CRL_CHECK_ALL)) != 0;
+    int crl_check_all_enabled =
+        (ctx->param->flags & X509_V_FLAG_CRL_CHECK_ALL) != 0;
+    int ocsp_check_enabled =
+        (ctx->param->flags &
+        (X509_V_FLAG_OCSP_RESP_CHECK | X509_V_FLAG_OCSP_RESP_CHECK_ALL)) != 0;
+
+    if (!crl_check_enabled && !ocsp_check_enabled)
         return 1;
-    if ((ctx->param->flags & X509_V_FLAG_CRL_CHECK_ALL) != 0) {
-        last = sk_X509_num(ctx->chain) - 1;
+
+    if (ocsp_check_enabled) {
+#ifndef OPENSSL_NO_OCSP
+        int ocsp_check_all_enabled =
+            (ctx->param->flags & X509_V_FLAG_OCSP_RESP_CHECK_ALL) != 0;
+
+        /*
+         * certificate status checking with OCSP
+         */
+        if (ocsp_check_all_enabled) {
+            last = sk_X509_num(ctx->chain) - 1;
+        } else {
+            last = 0;
+        }
+
+        for (i = 0; i <= last; i++) {
+            ctx->error_depth = i;
+            ctx->current_cert = sk_X509_value(ctx->chain, i);
+
+            if (ctx->current_cert->ex_flags & EXFLAG_SS)
+                continue;
```

Comment:
That's understood, but why do you do this?

Reply:
Right, but this check is not done here for CRLs either, simply because self-signed certs do not appear in the chain. They are used only as trust anchors.

Reply:
I will remove the check.

Reply:
After removing the statement the selfsigned root CA certificate was checked. The `ctx->parent == NULL` does not work in that case. So I will leave the check.

Reply:
Will add comment.